src/diffusers/pipelines/pyramid_broadcast_utils.py

```diff
@@ -120,40 +32,68 @@ def _enable_pyramid_attention_broadcast(self) -> None:
 
         for name, module in denoiser.named_modules():
             if isinstance(module, Attention):
-                logger.debug(f"Enabling Pyramid Attention Broadcast in layer: {name}")
+                is_spatial_attention = (
+                    any(x in name for x in self._pab_spatial_attn_layer_identifiers)
+                    and self._pab_spatial_attn_skip_range is not None
+                    and not module.is_cross_attention
```

Comment:
We could consider validating the inputs at the beginning when PAB is requested by the user w.r.t the known configurations. 

For example, if temporal attention is detected/requested on a module that is not of temporal attention, we error out with a helpful message.

Reply:
Not sure how we could know what layer of a module is specifically temporal attention without maintaining additional metadata per model (for example, for Latte it is "temporal_transformer_blocks", for AnimateDiff it is "motion_adapter", etc. - this info isn't known through the modeling code (...it is, but would require additional book-keeping which might get worse as more techniques are supported, so let's keep things simple and assume sensible defaults until a foolproof design can be converted to as more techniques are added), so I think good docs + sensible defaults should cover it for now).